analyses/cell-type-consensus/exploratory-notebooks/04-brain-and-cns-consensus-celltypes.Rmd

```diff
@@ -0,0 +1,403 @@
+
+```{r}
+# add column of "top cell types" for easier plotting 
+all_results_df <- all_results_df |> 
+  dplyr::mutate(
+    # get most frequently observed cell types across libraries in that project 
+    top_celltypes = forcats::fct_lump_n(consensus_annotation, 15, other_level = "All remaining cell types", ties.method = "first") |> 
+      # sort by frequency 
+      forcats::fct_infreq() |> 
+      # make sure all remaining and unknown are last, use this to assign colors in specific order
+      forcats::fct_relevel("All remaining cell types", "Unknown", after = Inf),
+  )
+```
+
+
+```{r}
+# get a vector of library ids ordered by total percentage annotated
+library_levels <- all_results_df |> 
+  dplyr::filter(consensus_annotation != "Unknown") |> 
+  dplyr::group_by(library_id) |> 
+  dplyr::summarize(
+    total_percent_annotated = sum(total_cells_per_annotation)/unique(total_cells_per_library)
+  ) |>
+  dplyr::arrange(desc(total_percent_annotated)) |> 
+  dplyr::pull(library_id)
+
+# append any libraries that have 0 cells annotated
+non_annotated_libs <- setdiff(unique(all_results_df$library_id), library_levels)
+library_levels <- c(library_levels, non_annotated_libs)
+
+# reorder by total percentage annotated 
+all_results_df <- all_results_df |> 
+  dplyr::mutate(
+    library_id = forcats::fct_relevel(library_id, library_levels)
+  ) |>
+  unique()
+```
+
+
+```{r}
+stacked_barchart(all_results_df, fill_column = "top_celltypes", celltype_colors = all_celltype_colors)
+```
+
+Below we show the same data but faceted by project. 
+
+```{r}
+stacked_barchart(all_results_df, fill_column = "top_celltypes", facet_variable = "project_id", celltype_colors = all_celltype_colors)
+```
+
+- Macrophages seem to be the dominant cell type across most samples. 
+- There appear to be some samples that have a higher proportion of T cells, but this is not project specific. 
+- As with the osteosarcoma samples there is some general heterogeneity both in the number of assigned cells and the cell types across all samples. 
+
+We hypothesize that there will be differences in the cell types observed based on the specific type of brain or CNS tumor. 
+Let's look at what tumor types are represented here and then use that to assign diagnosis groups so we can separate samples by diagnosis in the later plots. 
+
+```{r}
+all_results_df |>
+  dplyr::select(library_id, diagnosis) |> 
+  unique() |> 
+  dplyr::count(diagnosis, name = "number_of_samples") |>
+  dplyr::arrange(desc(number_of_samples))
+```
+
+Based on this let's make the following groups: 
+
+- Low-grade glioma: Any samples labeled with "Low-grade glioma" or are part of `SCPCP000002` which are categorized as LGG tumors. 
+- High-grade glioma: Any samples labeled with "High-grade glioma" or are part of `SCPCP000001` which are categorized as HGG tumors. 
```

Comment:
This should [include Diffuse midline glioma](https://www.cancer.gov/rare-brain-spine-tumor/tumors/diffuse-midline-gliomas) as well ([another source](https://radiopaedia.org/articles/diffuse-midline-glioma-h3-k27-altered?lang=us)).